programs/test/dlopen_demo.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+# Run the shared library dynamic loading demo program.
+# This is only expected to work when Mbed TLS is built as a shared library.
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e -u
+
+program_dir="${0%/*}"
+program="$program_dir/dlopen"
+top_dir="$program_dir/../.."
+library_dir="$top_dir/library"
+
+# ELF-based Unix-like (Linux, *BSD, Solaris, ...)
+if [ -n "${LD_LIBRARY_PATH-}" ]; then
+    LD_LIBRARY_PATH="$library_dir:$LD_LIBRARY_PATH"
+else
+    LD_LIBRARY_PATH="$library_dir"
+fi
+export LD_LIBRARY_PATH
+
+# OSX/macOS
+if [ -n "${DYLD_LIBRARY_PATH-}" ]; then
+    DYLD_LIBRARY_PATH="$library_dir:$DYLD_LIBRARY_PATH"
+else
+    DYLD_LIBRARY_PATH="$library_dir"
+fi
+export DYLD_LIBRARY_PATH
```

Comment:
How should I test this on macOS?

I get

```
mini1 $ git show | head -11
commit 7fb54c567456f3501ed711bbb72d1d2de2d130da
Author: Gilles Peskine <Gilles.Peskine@arm.com>
Date:   Wed Nov 10 21:04:24 2021 +0100

    More explicit output for the test program

    Without that, the logs were a bit hard to understand if you didn't know what
    to expect.

    Signed-off-by: Gilles Peskine <Gilles.Peskine@arm.com>

mini1 $ ls
3rdparty/		ChangeLog		Makefile.all.bak	configs/		library/
BRANCHES.md		ChangeLog.d/		README.md		dco.txt			programs/
BUGS.md			DartConfiguration.tcl	SECURITY.md		docs/			scripts/
CMakeLists.txt		LICENSE			SUPPORT.md		doxygen/		tests/
CONTRIBUTING.md		Makefile		cmake/			include/		visualc/

mini1 $ find . -name libmbedtls\*
./library/libmbedtls.dylib
./library/libmbedtls.a

mini1 $ sh programs/test/dlopen_demo.sh
Running dynamic loading test program: programs/test/dlopen
Loading libraries from: programs/test/../../library
Dynamic loading error for dlopen(libmbedtls.so): dlopen(libmbedtls.so, 2): image not found
```

Reply:
You tested it right. I'd only done half of the work required to support macOS. Fixed.